src/codeu/chat/server/BroadCastSystem.java

```diff
@@ -0,0 +1,148 @@
+
+        broadCaster = new Thread(messageBroadCaster);
+
+        broadCaster.start();
+    }
+
+    private void addConnection(Connection connection, Uuid uuid){
+
+        if (connection == null) throw new NullPointerException();
+        List<Connection> recipients = conversationUsers.get(uuid.id());
+        recipients.add(connection);
+
+    }
+
+    private void removeConnection(Connection connection, Uuid uuid){
+
+        if (connection == null) throw new NullPointerException();
+        List<Connection> recipients = conversationUsers.get(uuid.id());
+        recipients.remove(connection);
+
+    }
+
+    public void switchConversation(Connection connection, ConversationSummary oldCon, ConversationSummary newCon){
+
+        if (newCon == null && oldCon == null)
+            throw new NullPointerException();
+        if (oldCon != null)
+            removeConnection(connection, oldCon.id);
+        if (newCon != null)
+            addConnection(connection, newCon.id);
+
+    }
+
+    private void broadCastMessage(ConversationMessageLink messageLink){
+
+
+        int conversationId = messageLink.conversationUuid.id();
+        Message message = messageLink.message;
+
+        // using an iterator in order to remove connections if they return an exception
+        // this is in case the client has disconnected for whatever reason
+
+        Iterator<Connection> myIterator = conversationUsers.get(conversationId).iterator();
+
+        while (myIterator.hasNext()) {
+            Connection connection = myIterator.next();
+
+            try {
+
+                OutputStream out = connection.out();
+
+                synchronized (out) {
+                    Serializers.INTEGER.write(out, NetworkCode.NEW_BROADCAST);
+                    Message.SERIALIZER.write(out, message);
+                }
+
+            } catch (Exception ex) {
+                try {
+                    connection.close();
+                } catch (IOException ioexc) {
+                    System.out.println("Error closing socket");
+                }
+                myIterator.remove();
+            }
+
+        }
+
+    }
+
+    public void addConversation(ConversationSummary conversationSummary){
+
+        if (conversationSummary == null) throw new NullPointerException("Cannot add null to conversations");
+        if (conversationSummary.id == null) throw new NullPointerException("Null ID in conversation summary");
+
+        int uid = conversationSummary.id.id();
+
+        if (conversationUsers.contains(uid)) throw new IllegalArgumentException("Conversation already exists");
+
+        conversationUsers.put(uid, Collections.synchronizedList(new LinkedList<>()));
+
+    }
+
+    // adds the given message to the list of messages that need to be broadcasted
+    public void addMessage(Uuid conversationUuid, Message message){
+        if (message == null) throw new NullPointerException("Message cannot be null");
```

Comment:
So if user decides they would like to send " " for whatever reason I shouldn't push the message? 

The broadcast system, pushes any new messages that have been accepted by the server. As such, if I add the logic here, the empty message will still appear in the model. Would it be more appropriate to add such logic to the server, rather than the broadcaster?